lib/services/local/mfa.go

```diff
@@ -0,0 +1,172 @@
+			BackendPrefix: backend.NewKey(types.KindValidatedMFAChallenge),
+			MarshalFunc:   MarshalValidatedMFAChallenge,
+			UnmarshalFunc: UnmarshalValidatedMFAChallenge,
+			ValidateFunc:  validateValidatedMFAChallenge,
+		})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &MFAService{
+		logger: slog.With(teleport.ComponentKey, "MFAService.local"),
+		svc:    svc,
+	}, nil
+}
+
+// CreateValidatedMFAChallenge persists the ValidatedMFAChallenge resource.
+func (s *MFAService) CreateValidatedMFAChallenge(
+	ctx context.Context,
+	username string,
+	chal *mfav1.ValidatedMFAChallenge,
+) (*mfav1.ValidatedMFAChallenge, error) {
+	switch {
+	case ctx == nil:
+		return nil, trace.BadParameter("param ctx must not be nil")
+	case username == "":
+		return nil, trace.BadParameter("param username must not be empty")
+	case chal == nil:
+		return nil, trace.BadParameter("param chal must not be nil")
+	}
+
+	// Scope the service to the given username, so that the resource is created under that user's prefix.
+	// TODO(cthach): Copying can be expensive at scale, consult with team if there's a better way.
+	svc := s.svc.WithPrefix(username)
+
+	// All validated MFA challenges must expire after 5 minutes.
+	chal.Metadata.SetExpiry(time.Now().Add(5 * time.Minute))
+
+	res, err := svc.CreateResource(ctx, (*validatedMFAChallenge)(chal))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return (*mfav1.ValidatedMFAChallenge)(res), nil
+}
+
+// GetValidatedMFAChallenge retrieves a ValidatedMFAChallenge resource.
+func (s *MFAService) GetValidatedMFAChallenge(
+	ctx context.Context,
+	username string,
+	chalName string,
+) (*mfav1.ValidatedMFAChallenge, error) {
+	switch {
+	case ctx == nil:
+		return nil, trace.BadParameter("param ctx must not be nil")
+	case username == "":
+		return nil, trace.BadParameter("param username must not be empty")
+	case chalName == "":
+		return nil, trace.BadParameter("param chalName must not be empty")
+	}
+
+	// Scope the service to the given username, so that the resource is created under that user's prefix.
+	// TODO(cthach): Copying can be expensive at scale, consult with team if there's a better way.
+	svc := s.svc.WithPrefix(username)
+
+	res, err := svc.GetResource(ctx, chalName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return (*mfav1.ValidatedMFAChallenge)(res), nil
+}
+
+// validatedMFAChallenge is an alias to mfav1.validatedMFAChallenge in order to satisfy the generic.ResourceMetadata
+// interface. It should not be exported outside of this package to avoid confusion.
+type validatedMFAChallenge mfav1.ValidatedMFAChallenge
+
+func (r *validatedMFAChallenge) GetMetadata() *headerv1.Metadata {
+	return types.LegacyTo153Metadata(*r.Metadata)
+}
+
+// MarshalValidatedMFAChallenge marshals a ValidatedMFAChallenge resource into JSON.
+func MarshalValidatedMFAChallenge(chal *validatedMFAChallenge, opts ...services.MarshalOption) ([]byte, error) {
```

Comment:
Is it OK to ignore `opts`? Should we document that?

Should we forgo the param name so it's clear we are not using it?

```suggestion
func MarshalValidatedMFAChallenge(chal *validatedMFAChallenge, _ ...services.MarshalOption) ([]byte, error) {
```